frb_codegen/src/library/codegen/ir/mir/ty/delegate.rs

```diff
@@ -38,6 +38,7 @@ pub enum MirTypeDelegate {
     DynTrait(MirTypeDelegateDynTrait),
     Lifetimeable(MirTypeDelegateLifetimeable),
     CustomSerDes(MirTypeDelegateCustomSerDes),
+    Future(MirTypeDelegateFuture)
```

Comment:
Hmm a bit confused: we have both MirFuture as well as MirDelegate::Future. Seems we at most need one of them (the latter looks more lightweight).

Reply:
I've completely removed MirTypeFuture, and moved everything to MirTypeDelegateFuture. That way, there's little to no code generation changes needed

Reply:
Great! Do you want me to do a full review now, or wait for the PR to be not in "draft" status?